class.jetpack-calypsoify.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/*
+Plugin Name: Calypsoify
+Plugin URI: https://wordpress.com
+Description: WP-Admin skin and plugins only.
+Author: Automattic
+Author URI: https://automattic.com
+*/
+
+class Jetpack_Calypsoify {
+	static $instance = false;
+
+	private function __construct() {
+		add_action( 'wp_loaded', array( $this, 'setup' ) );
+	}
+
+	public static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function setup() {
+		if ( ! Jetpack::is_active() || ! Jetpack::is_module_active( 'masterbar' ) ) {
+			return;
+		}
+
+		add_action( 'admin_init', array( $this, 'check_param' ) );
+		if ( 1 == (int) get_user_meta( get_current_user_id(), 'calypsoify', true ) ) {
+			if ( $this->is_page_gutenberg() ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_for_gutenberg' ), 100 );
+				return;
+			}
+			add_action( 'admin_init', array( $this, 'check_page' ) );
+			add_action( 'admin_menu', array( $this, 'remove_core_menus' ), 100 );
+			add_action( 'admin_menu', array( $this, 'add_plugin_menus' ), 101 );
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue' ), 100 );
+			add_action( 'in_admin_header', array( $this, 'insert_sidebar_html' ) );
+			add_action( 'wp_before_admin_bar_render', array( $this, 'modify_masterbar' ), 100000 );
+		}
+	}
+
+	public function remove_core_menus() {
+		remove_menu_page( 'index.php' );
+		remove_menu_page( 'jetpack' );
+		remove_menu_page( 'edit.php' );
+		remove_menu_page( 'edit.php?post_type=feedback' );
+		remove_menu_page( 'upload.php' );
+		remove_menu_page( 'edit.php?post_type=page' );
+		remove_menu_page( 'edit-comments.php' );
+		remove_menu_page( 'themes.php' );
+		remove_menu_page( 'plugins.php' );
+		remove_menu_page( 'users.php' );
+		remove_menu_page( 'tools.php' );
+		remove_menu_page( 'link-manager.php' );
```

Comment:
Do we want to remove the Gutenberg menu?
![2018-10-09 at 15 59](https://user-images.githubusercontent.com/88897/46698336-61de4380-cbdc-11e8-8ecd-f790fe79b7c0.jpg)

If the intent is to mock Calypso, it feels weird to send folks to an editor that they can't use in Calypso through a normal flow. Folks would be literally stumbling into it as it stands now.

Reply:
Good question - Gutenberg will no longer be a plugin soon. Keep in mind this is all current behavior of live Atomic sites